tests/rptest/tests/acls_test.py

```diff
@@ -321,6 +321,55 @@ def test_mtls_principal(self, rules=None, fail=False):
             except ClusterAuthorizationError:
                 assert fail, "list acls should have succeeded"
 
+    @cluster(num_nodes=3)
+    @parametrize(authn_method='sasl')
+    @parametrize(authn_method='mtls_identity')
+    def test_security_feature_migration(self, authn_method: str):
+        def check_perms():
+            # run a few times for good health
+            for _ in range(2):
+                try:
+                    self.get_client("base").acl_list()
+                    raise RuntimeError(
+                        'list acls should have failed with base user')
+                except ClusterAuthorizationError:
+                    pass
+
+                # should succeed
+                self.get_client("cluster_describe").acl_list()
+                self.get_super_client().acl_list()
```

Comment:
this looks like a good candidate to control via `wait_until` helper. the reason being that propogating changes around the cluster might take some time so it is more of a eventually consistent type situation.

Reply:
Done.